dmithamo1/tbd
# Configure a Development Environment

#### What does this PR do?
It sets up a complete dev environment
#### Description of Task to be completed
- set up automated linting
- set up pre-commit hooks to enforce formatting and linting
- refine the pull request template

#### How should this be manually tested?
1. Clone the repo
2. Add a `.js` file in the `src` directory or change any of the ones already there.
3. Attempt to make a commit (i.e., `git add src/<nameOfFile.js>` then `git commit`).
Linting should be executed on all the project's files (in the `src` directory) before your commit can be made.
#### Any background context you want to provide?
Refer to [this article](https://blog.fossasia.org/integrate-prettier-with-lint-staged-and-eslint-for-consistent-code-style-throughout-the-project/) for more info on how to combine [husky](https://github.com/typicode/husky), [prettier](https://prettier.io/) and [eslint](https://eslint.org/). These are the main dependencies for this functionality.
#### Screenshots (if appropriate)

#### Checklist
